apm-agent-plugins/apm-log-correlation-plugin/src/main/java/co/elastic/apm/agent/mdc/MdcActivationListener.java

```diff
@@ -173,6 +177,7 @@ public void afterDeactivate(TraceContextHolder<?> deactivatedContext) throws Thr
                     if (tracer.getActive() == null) {
                         remove.invokeExact(TRACE_ID);
                         remove.invokeExact(TRANSACTION_ID);
+                        remove.invokeExact(ERROR_ID);
                     }
```

Comment:
to remove the `error.id` after capturing the error
```suggestion
                    if (context instanceof ErrorCapture) {
                        remove.invokeExact(ERROR_ID);
                    } else if (tracer.getActive() == null) {
                        remove.invokeExact(TRACE_ID);
                        remove.invokeExact(TRANSACTION_ID);
                    }
```

Reply:
fixed